ad9959/src/lib.rs

```diff
@@ -606,3 +699,35 @@ impl ProfileSerializer {
         bytemuck::cast_slice(&self.data[..self.index])
     }
 }
+
+/// Represents a fully defined DDS profile, with parameters expressed in machine units
+pub struct DdsProfile {
+    pub ftw: u32,
+    pub pow: u16,
+    pub acr: u32,
+}
+
+impl DdsProfile {
+    /// Construct a new DDS profile
+    ///
+    /// Args:
+    /// * `frequency` - The desired output frequency in Hz.
+    /// * `phase_turns` - The desired phase offset in turns.
+    /// * `amplitude` - A normalized amplitude setting [0, 1].
+    /// * `system_clock_frequency` - The system clock frequency of Ad9959.
+    pub fn new(
+        frequency: f32,
+        phase_turns: f32,
+        amplitude: f32,
+        system_clock_frequency: f32,
+    ) -> Result<Self, Error> {
+        Ok(DdsProfile {
+            ftw: Ad9959::<()>::frequency_to_ftw(
```

Comment:
So where should I be putting the converter methods (like `frequency_to_ftw`)? `Ad9959` is using them, otherwise I am duplicating all these code for the builder/serializer.

If we expose `Ad9959` back in `Pounder`, it sounds very much against the idea that we are freezing the `Ad9959` after destructing it and snatching the QSPI interface from it.